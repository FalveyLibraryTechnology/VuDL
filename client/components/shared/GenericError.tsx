import React from "react";
import Box from "@mui/material/Box";
import SentimentVeryDissatisfiedIcon from "@mui/icons-material/SentimentVeryDissatisfied";
import Link from "next/link";

interface GenericErrorProps {
    message: string;
}

const GenericError = ({ message }: GenericErrorProps): React.ReactElement => {
    return (
        <Box display="flex" flexDirection="row" justifyContent="center" alignItems="center" minHeight="100vh">
            <SentimentVeryDissatisfiedIcon sx={{ fontSize: "100px", color: "red" }} />
            <br />
            <Box width="300px">
                <Box sx={{ fontSize: "24px", padding: "4px" }}>
                    A generic error occurred. Please return <Link href="/">home.</Link>
                </Box>
                <Box sx={{ fontSize: "16px", padding: "4px" }}>Error details: {message}</Box>
            </Box>
        </Box>
    );
};

export default GenericError;
