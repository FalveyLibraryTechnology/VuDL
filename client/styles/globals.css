:root,
[color-scheme="light"] {
    --body-bg: #fafbfc;
    --heading-color: #4d4d4d;
    --link-color: #008aed;

    --btn-bg: #485a6c;
    --btn-text: #ffffff;
    --btn-border: #275575;
    --btn-focus-bg: #3498db;

    --btn-primary-bg: #27ae60;
    --btn-primary-text: #ffffff;
    --btn-primary-border: #275575;
    --btn-primary-focus-bg: #155724;

    --btn-danger-bg: red;
    --btn-danger-text: #ffffff;
    --btn-danger-border: #a10705;
    --btn-danger-focus-bg: #a10705;

    --btn-logout-bg: #1abc9c;
    --btn-logout-text: #ffffff;
    --btn-logout-border: #16a085;
    --btn-logout-focus-bg: #1abc9c;

    --btn-toggle-bg: #7f8c8d;
    --btn-toggle-text: #ffffff;
    --btn-toggle-border: #34495e;
    --btn-toggle-focus-bg: #53626f;

    --group-border: #53626f;

    --input-border: #6a737d;
    --input-focus-border: #2188ff;

    --label-hover: #c8e1ff;

    --page-list-bg: #ffffff;
    --page-list-border: #c1c1c1;

    --thumbnail-bg: #e1e1e1;
    --thumbnail-focus-bg: #51aded;
    --thumbnail-number-bg: #ffffff;
    --thumbnail-number-text: #000000;
    --thumbnail-preview-border: #000000;

    --tree-item-focus-bg: #000000;
    --tree-item-focus-text: #ffffff;

    --shadow-gray: #c8e1ff;
}

[color-scheme="dark"] {
}
