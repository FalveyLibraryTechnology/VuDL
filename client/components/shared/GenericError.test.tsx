import React from "react";
import { describe, expect, it } from "@jest/globals";
import renderer from "react-test-renderer";
import GenericError from "./GenericError";

describe("GenericError", () => {
    it("renders", () => {
        const message = "Test error";
        const tree = renderer.create(<GenericError message={message} />).toJSON();
        expect(tree).toMatchSnapshot();
        expect(JSON.stringify(tree).includes(message)).toBeTruthy();
    });
});
