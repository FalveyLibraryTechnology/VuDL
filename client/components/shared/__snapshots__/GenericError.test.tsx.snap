// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericError renders 1`] = `
<ForwardRef(Box)
  alignItems="center"
  display="flex"
  flexDirection="row"
  justifyContent="center"
  minHeight="100vh"
>
  <Memo(ForwardRef(SentimentVeryDissatisfiedIcon))
    sx={
      Object {
        "color": "red",
        "fontSize": "100px",
      }
    }
  />
  <br />
  <ForwardRef(Box)
    width="300px"
  >
    <ForwardRef(Box)
      sx={
        Object {
          "fontSize": "24px",
          "padding": "4px",
        }
      }
    >
      A generic error occurred. Please return 
      <Link
        href="/"
      >
        home.
      </Link>
    </ForwardRef(Box)>
    <ForwardRef(Box)
      sx={
        Object {
          "fontSize": "16px",
          "padding": "4px",
        }
      }
    >
      Error details: 
      Test error
    </ForwardRef(Box)>
  </ForwardRef(Box)>
</ForwardRef(Box)>
`;
